Index: app/helpers/xml_helper.rb
===================================================================
--- app/helpers/xml_helper.rb	(revision 1543)
+++ app/helpers/xml_helper.rb	(working copy)
@@ -40,10 +40,12 @@
   ######
 
   def load_xml(xml)
-    require 'xml/libxml'
+    require 'hpricot'
+    require 'rexml/document'
     begin
       raise if xml.empty?
-      @this_thing = XML::Parser.string(xml).parse
+      REXML::Document.new(xml)
+      @this_thing = Hpricot.XML(xml)
       return true
     rescue
       return false
@@ -55,7 +57,7 @@
       true
     else
       false
-    end    
+    end
   end
   def xpath_foreach(xpath_thing)
     for this_thing in get_matches(xpath_thing) 
@@ -92,19 +94,23 @@
   #######
 
   def get_matches(what_to_match)
-    @this_thing.find(what_to_match).to_a
+    @this_thing.search(what_to_match).to_a
   end
   def remove_tag(s, tag)
     s.to_s.gsub(/<#{tag}\b[^>]*>(.*?)<\/#{tag}>/, '\1')
   end
   def remove_attribute(s, attr)
-    s.to_s.gsub(/#{attr}=(["'])([^\1]*)\1/, '\2')
+#    s.to_s.gsub(/#{attr}=(["'])([^\1]*)\1/, '\2')
+    x = s.to_s.match(/#{attr}=(["'])([^\1]*)\1/).to_a[2]
+#    x = s.to_s.match("#{attr}=\"([^\"]*)\"").to_a[1]
+#    return (x || "")
   end
   def whats_in_this_thing(what_to_get)
     nodes=get_matches(what_to_get).first
-    what_was_got = File.basename(what_to_get)
+#    what_was_got = File.basename(what_to_get)
+what_was_got = ""
     if what_was_got[0] == '@'[0]
-      remove_attribute(nodes, what_was_got.sub(/@/, ""))
+      remove_attribute(nodes, what_was_got)
     else
       remove_tag(nodes, what_was_got)
     end
Index: app/views/spec_sheets/_memory.html.erb
===================================================================
--- app/views/spec_sheets/_memory.html.erb	(revision 1543)
+++ app/views/spec_sheets/_memory.html.erb	(working copy)
@@ -1,6 +1,6 @@
   <div id="memory">
 	<h2>Memory</h2>
-	<% xpath_foreach("//node[contains(@id, 'memory')]") do %>
+	<% xpath_foreach("//node[@id*='memory']") do %>
 	<% if xpath_if("size") %>
 	<table>
 	  <tr>
@@ -13,7 +13,7 @@
 	  </tr>
 	</table>
 	<% end %>
-	<% xpath_foreach("node[contains(@id, 'bank')]") do %>
+	<% xpath_foreach("node[@id*='bank']") do %>
 	
 	<h3>
 	  <%= xpath_value_of("@id") %>
Index: app/views/spec_sheets/_pci.html.erb
===================================================================
--- app/views/spec_sheets/_pci.html.erb	(revision 1543)
+++ app/views/spec_sheets/_pci.html.erb	(working copy)
@@ -3,8 +3,8 @@
     <% xpath_foreach("//node[contains(@id, 'pci') or contains(@id, 'pcmcia')]") do %>
     <h3><%= xpath_value_of("product") %></h3>
     <% xpath_foreach("node[
-                      (contains(@handle, 'PCI:')
-                      or contains(@handle, 'PCMCIA:'))
+                      (@handle*='PCI:'
+                      or @handle*='PCMCIA:')
                       and not(@class='bus')
                       and not(@class='bridge')
                       and not(@class='storage')
Index: app/views/spec_sheets/show.html.erb
===================================================================
--- app/views/spec_sheets/show.html.erb	(revision 1543)
+++ app/views/spec_sheets/show.html.erb	(working copy)
@@ -6,7 +6,7 @@
 <%= render :partial => "gizmo" %>
 <%= render :partial => "disburse" %>
 <%= render :partial => "harddrives" %>
-</div> <% #left %>
+</div>
 <div id="right" class="right">
 <%= render :partial => "memory" %>
 <%= render :partial => "optical" %>
Index: app/views/spec_sheets/_optical.html.erb
===================================================================
--- app/views/spec_sheets/_optical.html.erb	(revision 1543)
+++ app/views/spec_sheets/_optical.html.erb	(working copy)
@@ -1,6 +1,6 @@
   <div id="optical_drives">
     <h2>Optical Drives</h2>
-    <% xpath_foreach("//node[contains(@id, 'cdrom')]") do %>
+    <% xpath_foreach("//node[@id*='cdrom']") do %>
     <table>
       <tr>
         <td>Drive:</td>
@@ -26,4 +26,4 @@
       </tr>
     </table>
     <% end %>
-  </div> <% #optical_drives %>
+  </div> 
Index: app/views/spec_sheets/_harddrives.html.erb
===================================================================
--- app/views/spec_sheets/_harddrives.html.erb	(revision 1543)
+++ app/views/spec_sheets/_harddrives.html.erb	(working copy)
@@ -1,6 +1,6 @@
   <div id="hard_drives">
     <h2>Hard Drives</h2>
-    <% xpath_foreach("//node[contains(@id, 'disk')]") do %>
+    <% xpath_foreach("//node[@id*='disk']") do %>
     <table>
       <tr>
         <td>
@@ -36,12 +36,12 @@
         </td>
       </tr>
       <% end %>
-      <% if xpath_if("node[contains(@id, 'volume')]") %>
+      <% if xpath_if("node[@id*='volume']") %>
       <tr>
         <td valign="top">Volumes: </td>
         <td>
           <table>
-            <% xpath_foreach("node[contains(@id, 'volume')]") do %>
+            <% xpath_foreach("node[@id*='volume']") do %>
             <tr>
               <td>
                 <%= xpath_value_of("logicalname") %>
Index: app/views/spec_sheets/_computer.html.erb
===================================================================
--- app/views/spec_sheets/_computer.html.erb	(revision 1543)
+++ app/views/spec_sheets/_computer.html.erb	(working copy)
@@ -55,7 +55,7 @@
         <%= xpath_value_of("//node[@id='firmware']/version") %>
       </td>
     </tr>
-    <% find_the_biggest("//node[contains(@id, 'usbhost')]", "capabilities/capability", "USB ") do |value| %>
+    <% find_the_biggest("//node[@id*='usbhost']", "capabilities/capability", "USB ") do |value| %>
     <tr>
       <td>
         USB Version:
